### PR TITLE
[terminal] Added basic link matching

### PR DESCRIPTION
**Pull request by svenefftinge**
See theia-ide/theia#3724

Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

**Comment by svenefftinge** [comment]
I will address #3724 in a separate PR as it will require a CQ.

File: packages/terminal/src/browser/terminal-linkmatcher.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { ILinkMatcherOptions } from "xterm";
```
Comment (AlexTugarev) [inline packages/terminal/src/browser/terminal-linkmatcher.ts]: ' 
:-(

**Comment by AlexTugarev** [comment]
This worked nicely for me :-)

Some remarks:
If the Terminal wasn't focused before, and you click a link, it wont receive focus.
And WDYT of link activation with a modifier?

File: packages/terminal/src/browser/terminal-linkmatcher.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    handler(event: MouseEvent, uri: string): void;
+}
+
+@injectable()
+export class URLMatcher implements ITerminalLinkMatcher {
+
+    getRegex() { return /(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/; }
+
+    handler(event: MouseEvent, uri: string): void {
+        window.open(uri, '_blank');
```
Comment (AlexTugarev) [inline packages/terminal/src/browser/terminal-linkmatcher.ts]: I think we should use `WindowService.openNewWindow` here to support Electron properly. Same applies for `LocalhostMatcher`

**Comment by svenefftinge** [comment]
@AlexTugarev I added the cmd+click behavior

**Comment by AlexTugarev** [comment]
testing again...

**Comment by AlexTugarev** [comment]
@svenefftinge, awesome! `cmd+click` works nice and I think it's good to avoid accidental jumps, esp. when you try to select the url.

there is a missing `;`, otherwise 🛫

**Pull request merged**